src/sim_engine/filesystem.rs

```diff
@@ -6,15 +6,29 @@ use engine::Filesystem;
 
 #[derive(Debug, Clone)]
 pub struct SimFilesystem {
+    pub name: String,
```

Comment:
Could we leave out name field as it is redundant because we already map name to fileystem object.